### PR TITLE
New Feature: Application Boxes in Indexer

### PR DESCRIPTION
**Pull request by tzaffi**
Integration branch where "ready to merge" box-related code will live until `go-algorand` box code is itself merged.

# TODO before merging

- [x] Has go-algorand's [feature/avm-box](https://github.com/algorand/go-algorand/pull/4149) been merged?
- [x] End-to-end test
  - [x] is passing against network that includes boxes
  - [x] has been re-pointed to `rel-nightly` once that job includes boxes
  - [x] `.circleci/config.yml`'s `CI_E2E_FILENAME` has been reverted
- [x] Re-point the `third_party/go-algorand` submodule to 
  - [x] `go-algorand` 
  - [x] a released commit containing boxes

### CODE REVIEW

File: misc/docker-compose.yml
Comment (tzaffi) [inline misc/docker-compose.yml:10]: ^^^ TODO: Revert before merging ^^^

File: idb/postgres/postgres_integration_test.go
Comment (tzaffi) [inline idb/postgres/postgres_integration_test.go:2104]:
```suggestion
// Test that if genesis hash is different from what is in db metastate
```

File: misc/e2elive.py
Comment (tzaffi) [inline misc/e2elive.py:92]:
```suggestion
            s3, bucket, prefix, tarname, outpath=tarpath, max_s3_keys=50
```
TODO: revert to original value before merging.

File: misc/docker-compose.yml
Comment (tzaffi) [inline misc/docker-compose.yml:10]:
```suggestion
      CI_E2E_FILENAME: "rel-nightly"
```

^^^ TODO: revert before merging ^^^

File: .circleci/config.yml
Comment (tzaffi) [inline .circleci/config.yml:49]:
```suggestion
      CI_E2E_FILENAME: "fafa8862/rel-nightly"
```
Don't merge before reverting

File: util/test/programs.go
Comment (winder) [inline util/test/programs.go:4]: nit: this is a good candidate for `go embed` https://pkg.go.dev/embed

**Review by winder — APPROVED** [review]
LGTM, I think we just need the algod side now.

File: .circleci/config.yml
Comment (tzaffi) [inline .circleci/config.yml:49]:
```suggestion
      CI_E2E_FILENAME: "feature-avm-box"
```
Before merging: revert this to a more up to date E2E tarball that is from `go-algorand/master` but includes boxes

File: util/test/programs.go
Comment (tzaffi) [reply util/test/programs.go:4]: resolving in favor of #1227

File: idb/postgres/postgres_boxes_test.go
Comment (algochoi) [inline idb/postgres/postgres_boxes_test.go:462]:
```suggestion
	fmt.Printf("TestRandomWriteReadBoxes total time: %s\n", time.Since(start))
```

File: api/handlers.go
Comment (algochoi) [inline api/handlers.go:644]:
```suggestion
			return badRequest(ctx, fmt.Sprintf("SearchForApplicationBoxes received illegal next token (%s): %s", encodedBoxName, err.Error()))
```

File: idb/postgres/postgres.go
Comment (tzaffi) [reply idb/postgres/postgres.go:2328]: Currently it is inaccessible externally. I.e., results are always "ASC" and so L 2329 is unreachable.

File: processor/eval/ledger_for_evaluator_test.go
Comment (tzaffi) [reply processor/eval/ledger_for_evaluator_test.go]: deleted

File: idb/postgres/postgres.go
Comment (algochoi) [reply idb/postgres/postgres.go:2328]: Do we ever want to sort (or give the option to) descending? 

If we sort by descending, do we want to change this line (a couple lines above this one) to be `name < $2`
https://github.com/algorand/indexer/blob/32df6f8d530354d0c9e4b0d5018cf9fa87ee6042/idb/postgres/postgres.go#L2323

Comment (tzaffi) [reply idb/postgres/postgres.go:2328]: This is a good point. I'm going to refactor to slim down the query code and option struct, and hard-code ordering.

File: .circleci/config.yml
Comment (tzaffi) [inline .circleci/config.yml:48]:
```suggestion
    environment:
```

**Review by tzaffi — COMMENTED** [review]
spurious space

**Pull request merged**
